src/sagemaker/amazon/ipinsights.py

```diff
@@ -0,0 +1,143 @@
+
+        IPInsights Estimators can be configured by setting hyperparamters.
+        The available hyperparamters are documented below.
+
+        For further information on the AWS IPInsights algorithm, please consult AWS technical documentation:
+        https://docs.aws.amazon.com/sagemaker/latest/dg/ip-insights-hyperparameters.html
+
+        Args:
+            role (str): An AWS IAM role (either name or full ARN). The Amazon SageMaker training jobs and
+                APIs that create Amazon SageMaker endpoints use this role to access
+                training data and model artifacts. After the endpoint is created,
+                the inference code might use the IAM role, if accessing AWS resource.
+            train_instance_count (int): Number of Amazon EC2 instances to use for training.
+            train_instance_type (str): Type of EC2 instance to use for training, for example, 'ml.m5.xlarge'.
+            num_entity_vectors (int): Required. The number of embeddings to train for entities accessing online
+                resources. We recommend 2x the total number of unique entity IDs.
+            vector_dim (int): Required. The size of the embedding vectors for both entity and IP addresses.
+            batch_metrics_publish_interval (int): Optional. The period at which to publish metrics (batches).
+            epochs (int): Optional. Maximum number of passes over the training data.
+            learning_rate (float): Optional. Learning rate for the optimizer.
+            num_ip_encoder_layers (int): Optional. The number of fully-connected layers to encode IP address embedding.
+            random_negative_sampling_rate (int): Optional. The ratio of random negative samples to draw during training.
+                Random negative samples are randomly drawn IPv4 addresses.
+            shuffled_negative_sampling_rate (int): Optional. The ratio of shuffled negative samples to draw during
+                training. Shuffled negative samples are IP addresses picked from within a batch.
+            weight_decay (float): Optional. Weight decay coefficient. Adds L2 regularization.
+            **kwargs: base class keyword argument values.
+        """
+
+        super(IPInsights, self).__init__(role, train_instance_count, train_instance_type, **kwargs)
+        self.num_entity_vectors = num_entity_vectors
+        self.vector_dim = vector_dim
+        self.batch_metrics_publish_interval = batch_metrics_publish_interval
+        self.epochs = epochs
+        self.learning_rate = learning_rate
+        self.num_ip_encoder_layers = num_ip_encoder_layers
+        self.random_negative_sampling_rate = random_negative_sampling_rate
+        self.shuffled_negative_sampling_rate = shuffled_negative_sampling_rate
+        self.weight_decay = weight_decay
+
+    def create_model(self, vpc_config_override=VPC_CONFIG_DEFAULT):
+        """Return a :class:`~sagemaker.amazon.IPInsightsModel`referencing the latest
```

Comment:
this can go in a `Returns` section after `Args`. also, add a space after the last backtick

Reply:
Done